client/js/store/statement/Statement.js

```diff
@@ -89,6 +89,43 @@ function getDataFromResponse (response, dataToUpdate, updatedData) {
   return dataToUpdate
 }
 
+function extractFileAttachments (attachments, includes, { single = false } = {}) {
+  const results = attachments
+    .map(att => {
+      const fileId = att?.relationships?.file?.data?.id
+      const file = includes.find(include => include.type === 'File' && include.id === fileId)
+
+      return file ? { id: file.id, ...file.attributes } : null
+    })
+    .filter(Boolean)
+
+  if (single) {
+    return results[0] || null
+  }
+
+  return results
+}
+
+function mapRelations (data, includes) {
+  const ids = data.map(el => el.id)
+  const type = data[0].type
+
+  return includes
+    .filter(item => item.type === type && ids.includes(item.id))
+    .map(item => ({
+      id: item.id,
+      ...item.attributes,
+      ...(item.relationships && { relationships: item.relationships })
+    }))
+}
+
+function mapSingleRelation (data, includes) {
```

Comment:
i suggest to use the spreadOpperator to have it straight in view what data can hold:
```
function mapSingleRelation ({ id }, includes) {
```